### ISSUE TITLE
RangeSelector - screenreader issues

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7027]
### Expected Behavior
When navigating with a keyboard and using a screenreader the screenreader should read out the value of the RangeSelector as it changes.

### Actual Behavior
Instead the screenreader reads out the previous value when the RangeSelector changes.

### URL, screen shot, or Codepen exhibiting the issue
https://storybook.grommet.io/?path=/story/input-rangeselector-label--label

### Steps to Reproduce

1. Go to the story above
2. Using a screenreader and keyboard change the value of the RangeSelector

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by KushalNerella07** [comment]
Hi, I'd like to tackle the RangeSelector screenreader issue to ensure it reads the current value accurately. May I work on this? Thanks!

**Comment by jcfilben** [comment]
Hi @KushalNerella07 feel free to work on this!

**Comment by abdulbasithqb** [comment]
i checked with Voiceover, seems like working as expected, @jcfilben could you please confirm ?

**Comment by jcfilben** [comment]
When I tested with voiceover the screenreader was announcing the previous value rather than the current value. For example when I increase the value from '3' to '4' the screenreader reads out '3'.

**Comment by jcfilben** [comment]
Looking at APG's example their slider has the same voiceover behavior as the Grommet one so maybe this is the expected behavior? It seems like it would be confusing to screenreader users though.

**For the grommet team, we should check with Bill T. on this to understand what the expected behavior is here.**

**Comment by abdulbasithqb** [comment]
> hen I increase the value from '3' to '4'
Just wondering which browser you are using. For me, it reads the current value as expected. I'm using Chrome as a browse

https://github.com/grommet/grommet/assets/134603758/18e9fa86-aa21-48ea-82ea-0809c1098ea0

**Comment by jcfilben** [comment]
Hmm interesting I am using voiceover with chrome as well. @grommet/devs is anyone else to check on their end if voiceover is reading out the previous value or current value for the RangeSelector?

**Comment by halocline** [comment]
> Hmm interesting I am using voiceover with chrome as well. @grommet/devs is anyone else to check on their end if voiceover is reading out the previous value or current value for the RangeSelector?

I noticed that VoiceOver seemed to be reading the current value. This was in a DataFilters `layer` context. Have not tested alone.

Chrome Version 122.0.6261.94 (Official Build) (x86_64)
MacOS Ventura v13.6.1

**Comment by halocline** [comment]
I just tested on Safari, FF, and Chrome and am getting inconsistent results for all.

Occasionally Chrome behaves correctly, but hard to reproduce. Typically it announces the current value.

Safari does not maintain focus on current slider when using arrow keys. Jumps to next focusable element.

FF jumps between values inconsistently.

One thing I thought I noticed was how using ctrl+option+shift+down arrow to enter the slider triggered a step increment / decrement.

**Comment by jcfilben** [comment]
Confirm if this issue is still a problem. Test in multiple browsers

**Comment by britt6612** [comment]
I tested and saw the correct behavior in both FF as well as Chrome.

There is still an issue in the Safari,
Safari with VoiceOver has known limitations when handling custom slider components that are built using `div` or other non-native elements, even when they correctly use:

`role="slider"`
`aria-valuemin`
`aria-valuemax`
`aria-valuenow`
`aria-valuetext`

The custom slider often fails to announce correctly. It is announcing phrases like "new line" instead of slider updates.
Safari’s accessibility tree does not fully support ARIA `role="slider"` on non-native elements.
VoiceOver in Safari heavily prefers native form controls (like <input type="range" />) for sliders

https://www.w3.org/WAI/ARIA/apg/patterns/slider/
https://stackoverflow.com/questions/55038961/buttons-within-a-slider-are-not-called-out-from-voiceover-a11y
https://github.com/leongersen/noUiSlider/issues/802
